src/platform_impl/linux/wayland/scale.rs

```diff
@@ -0,0 +1,52 @@
+use std::ops::Mul;
+
+use dpi::{LogicalSize, PhysicalSize};
+
+/// A wp-fractional-scale scale.
+///
+/// This type implements the `physical_size = round_half_up(logical_size * scale)`
+/// operation with infinite precision as required by the protocol.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct Scale {
+    scale: u64,
+}
+
+const BASE: u32 = 120;
+const BASE_U64: u64 = BASE as u64;
+const BASE_F64: f64 = BASE as f64;
+
+impl Scale {
+    pub fn from_wp_fractional_scale(v: u32) -> Self {
+        assert!(v > 0);
+        Self { scale: v as u64 }
+    }
+
+    pub fn from_integer_scale(v: u32) -> Self {
+        Self::from_wp_fractional_scale(v.saturating_mul(BASE))
+    }
+
+    pub fn to_f64(self) -> f64 {
+        self.scale as f64 / BASE_F64
+    }
+
+    pub fn round_up(self) -> u32 {
```

Comment:
Do we really need a roundup? The old logic was `trunc`. Though, the path were it can happen is always having an integer, so it would never round in reality.